iteration/iteration_test.go

```diff
@@ -38,289 +38,273 @@ func (s *TestIterationRepository) TearDownTest() {
+		require.Nil(t, err)
+		require.NotEqual(t, uuid.Nil, i3.ID)
+	})
 }
 
-func (s *TestIterationRepository) TestListIterationBySpace() {
+func (s *TestIterationRepository) TestLoad() {
 	t := s.T()
 	resource.Require(t, resource.Database)
 
 	repo := iteration.NewIterationRepository(s.DB)
-
-	fxt := tf.NewTestFixture(s.T(), s.DB, tf.Spaces(2))
-
-	for i := 0; i < 3; i++ {
+	t.Run("load root iteration", func(t *testing.T) {
 		start := time.Now()
 		end := start.Add(time.Hour * (24 * 8 * 3))
-		name := "Sprint #2" + strconv.Itoa(i)
+		name := "Sprint #24"
+		name2 := "Sprint #24.1"
+
+		fxt := tf.NewTestFixture(s.T(), s.DB, tf.Spaces(1))
 
 		i := iteration.Iteration{
 			Name:    name,
 			SpaceID: fxt.Spaces[0].ID,
 			StartAt: &start,
 			EndAt:   &end,
 		}
-		e := repo.Create(context.Background(), &i)
-		require.Nil(t, e)
-	}
-	// add iteration to the second space from our fixture
-	e := repo.Create(context.Background(), &iteration.Iteration{
-		Name:    "Other Spring #2",
-		SpaceID: fxt.Spaces[1].ID,
+		repo.Create(context.Background(), &i)
+
+		parentPath := append(i.Path, i.ID)
+		require.NotNil(t, parentPath)
+		i2 := iteration.Iteration{
+			Name:    name2,
+			SpaceID: fxt.Spaces[0].ID,
+			StartAt: &start,
+			EndAt:   &end,
+			Path:    parentPath,
+		}
+		repo.Create(context.Background(), &i2)
+
+		res, err := repo.Root(context.Background(), fxt.Spaces[0].ID)
+		require.Nil(t, err)
+		assert.Equal(t, i.Name, res.Name)
+		assert.Equal(t, i.ID, res.ID)
+		expectedPath := i.Path.Convert()
+		require.NotNil(t, res)
+		assert.Equal(t, expectedPath, res.Path.Convert())
 	})
-	require.Nil(t, e)
 
-	its, err := repo.List(context.Background(), fxt.Spaces[0].ID)
-	assert.Nil(t, err)
-	assert.Len(t, its, 3)
-}
+	t.Run("list by space", func(t *testing.T) {
+		fxt := tf.NewTestFixture(s.T(), s.DB, tf.Spaces(2))
 
-func (s *TestIterationRepository) TestUpdateIteration() {
-	t := s.T()
-	resource.Require(t, resource.Database)
-	repo := iteration.NewIterationRepository(s.DB)
+		for i := 0; i < 3; i++ {
+			start := time.Now()
+			end := start.Add(time.Hour * (24 * 8 * 3))
+			name := "Sprint #2" + strconv.Itoa(i)
 
-	start := time.Now()
-	end := start.Add(time.Hour * (24 * 8 * 3))
+			i := iteration.Iteration{
+				Name:    name,
+				SpaceID: fxt.Spaces[0].ID,
+				StartAt: &start,
+				EndAt:   &end,
+			}
+			e := repo.Create(context.Background(), &i)
+			require.Nil(t, e)
+		}
+		// add iteration to the second space from our fixture
+		e := repo.Create(context.Background(), &iteration.Iteration{
+			Name:    "Other Spring #2",
+			SpaceID: fxt.Spaces[1].ID,
+		})
+		require.Nil(t, e)
 
-	fxt := tf.NewTestFixture(s.T(), s.DB,
-		tf.Iterations(1,
-			tf.UserActive(false),
-			func(fxt *tf.TestFixture, idx int) error {
+		its, err := repo.List(context.Background(), fxt.Spaces[0].ID)
+		assert.Nil(t, err)
+		assert.Len(t, its, 3)
+	})
+
+	t.Run("load children", func(t *testing.T) {
+		fxt := tf.NewTestFixture(s.T(), s.DB,
```

Comment:
this test seems to verify more than 1 thing. Could you split it into subtests to make it simpler to understand ?

Reply:
2 subtests, one for success and one for failure, done in https://github.com/fabric8-services/fabric8-wit/pull/1719/commits/97492ca5a918441e492cd3780e706b2eedcec860